doc/source/namespaces.rst

```diff
@@ -0,0 +1,83 @@
+.. _namespaces-guide:
+
+Using Namespaces
+================
+
+A namespace is a logical grouping of jobs and named actors. When an actor is
+named, its name must be unique within the namespace.
+
+In order to set your applications namespace, it should be specified when you
+first connect to the cluster.
+
+.. code-block:: python
+
+   ray.init(namespace="hello")
+   # or using ray client
+   ray.util.connect(..., namespace="world")
+
+Named actors are only accessible within their namespaces.
+
+.. code-block:: python
+
+    import
+
+    @ray.remote
+    class Actor:
+      pass
+
+    # Job 1 creates two actors, "orange" and "purple" in the "colors" namespace.
+    ray.util.connect("", namespace="colors")
+    Actor.options(name="orange", lifetime="detached")
+    Actor.options(name="purple", lifetime="detached")
+    ray.util.disconnect()
+
+    # Job 2 is now connecting to a different namespace.
+    ray.util.connect("", namespace="fruits")
+    # This fails because "orange" was defined in the "colors" namespace.
+    ray.get_actor("orange")
+    # This succceeds because the name "orange" is unused in this namespace.
+    Actor.options(name="orange", lifetime="detached")
+    Actor.options(name="watermelon", lifetime="detached")
+    ray.util.disconnect()
+
+    # Job 3 connects to the original "colors" namespace
+    ray.util.connect("", namespace="colors")
+    # This fails because "watermelon" was in the fruits namespace.
+    ray.get_actor("watermelon")
+    # This returns the "orange" actor we created in the first job, not the second.
+    ray.get_actor("orange")
+    ray.util.disconnect()
+         
+
+Anonymous namespaces
+--------------------
+
+When a namespace is not specified, Ray will place your job in an anonymous
```

Comment:
Thought we decided to hold this off in the short-term? Did we change that decision?

Reply:
I think we've decided to do it cc @pcmoritz

Reply:
Btw, it will break all detached actors code (e.g., Airflow). Maybe creating a task to print proper warning in this case with a high priority (so that we won't forget about it)?

Reply:
We also need to update the detached actor section of the document.

Reply:
The named actors part is updated (see docs: https://ray--15774.org.readthedocs.build/en/15774/actors.html#named-actors) . The warning is already there (see `test_detached_warning` )

Reply:
Nice. It makes sense to raise warning when detached actors are used with anonymous namespace. 

Do you mind updating the doc in a little more detail? Probably just add one more line like this; 

```
# Create an actor with a name
counter = Counter.options(name="some_name").remote()

Separately, actor lifetimes can be decoupled from the job, allowing an actor to persist even after the driver process of the job exits.

counter = Counter.options(name="CounterActor", lifetime="detached").remote()
The CounterActor will be kept alive even after the driver running above script exits. Therefore it is possible to run the following script in a different driver: # -> Add one more line here that the new script should ensure the same namespace.

counter = ray.get_actor("CounterActor")
print(ray.get(counter.get_counter.remote()))
Note that the lifetime option is decoupled from the name. If we only specified the name without specifying lifetime="detached", then the CounterActor can only be retrieved as long as the original driver is still running.